### DISCUSSION TITLE
How to insert a blank line?

### DISCUSSION THREAD

**Discussion by andreas-vester** [mkdocstrings/mkdocstrings#493] [Q&A] [answered]
**Add detailed information, like**
- *mkdocstrings* version: 0.19.0

How can I put some text in my docstring to a new line?

```
def docstest(int_val: int) -> int:
    """Function to showcase docstring problems.

    Parameters
    ----------
    int_val
        Integer value.
        This text should be in a new line. How can I do that?

    Returns
    -------
    int:
        Returns an integer value

    """
    pass
```

![grafik](https://user-images.githubusercontent.com/29901736/196045631-dbf75d31-41b1-4777-a119-80a402208e2d.png)

**Comment by pawamoy** [comment]
Two solutions:

- add two trailing spaces to the previous line
- use `<br>` to insert line breaks

**Comment by andreas-vester** [comment]
> Two solutions:
> 
>     * add two trailing spaces to the previous line
Trailing spaces will automatically be removed by using the ``black`` formatter.
>     * use `<br>` to insert line breaks
This seems to be working, however, it pollutes the docstring. For example, using ``help(doctest)`` on the command line will now reveal the ``<br>`` command.

Inserting a blank line in the docstring seems to be translated to a regular new line when generating the online docs using ``mkdocstrings``.

**Comment by pawamoy** [comment]
> Trailing spaces will automatically be removed by using the black formatter.

Could you see if there's an issue for this on their tracker? And otherwise open one?

**Comment by pawamoy** [comment]
Found one: https://github.com/psf/black/issues/3306

**Comment by pawamoy** [comment]
The CommonMark spec says that we can use a backslash for hard line breaks. Unfortunately it seems that Python-Markdown does not support it.

**Reply by pawamoy** [reply, marked as answer]
It has been asked before: https://github.com/Python-Markdown/markdown/issues/1157.
Suggested solution: use the `pymdownx.escapeall` extension, with the `hardbreak` option enabled:

```yaml
markdown_extensions:
- pymdownx.escapeall:
    hardbreak: yes
```
